src/libraries/System.Private.CoreLib/src/System/IndexOfAnyAsciiSearcher.cs

```diff
@@ -0,0 +1,333 @@
+            {
+                ref short twoVectorsAfterStart = ref Unsafe.Add(ref searchSpace, 2 * Vector128<short>.Count);
+
+                do
+                {
+                    currentSearchSpace = ref Unsafe.Subtract(ref currentSearchSpace, 2 * Vector128<short>.Count);
+
+                    Vector128<short> source0 = Vector128.LoadUnsafe(ref currentSearchSpace);
+                    Vector128<short> source1 = Vector128.LoadUnsafe(ref currentSearchSpace, (nuint)Vector128<short>.Count);
+
+                    Vector128<byte> result = IndexOfAnyLookup<TNegator, TOptimizations>(source0, source1, bitmap);
+                    if (result != Vector128<byte>.Zero)
+                    {
+                        return ComputeLastIndex<TNegator>(ref searchSpace, ref currentSearchSpace, result);
+                    }
+                }
+                while (Unsafe.IsAddressGreaterThan(ref currentSearchSpace, ref twoVectorsAfterStart));
+            }
+
+            // We have 1-16 characters remaining. Process the first and last vector in the search space.
+            // They may overlap, but we'll handle that in the index calculation if we do get a match.
+            {
+                ref short oneVectorAfterStart = ref Unsafe.Add(ref searchSpace, Vector128<short>.Count);
+
+                ref short secondVector = ref Unsafe.IsAddressGreaterThan(ref currentSearchSpace, ref oneVectorAfterStart)
+                    ? ref Unsafe.Subtract(ref currentSearchSpace, Vector128<short>.Count)
+                    : ref searchSpace;
+
+                Vector128<short> source0 = Vector128.LoadUnsafe(ref searchSpace);
+                Vector128<short> source1 = Vector128.LoadUnsafe(ref secondVector);
+
+                Vector128<byte> result = IndexOfAnyLookup<TNegator, TOptimizations>(source0, source1, bitmap);
+                if (result != Vector128<byte>.Zero)
+                {
+                    return ComputeLastIndexOverlapped<TNegator>(ref searchSpace, ref secondVector, result);
+                }
+            }
+
+            return -1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> IndexOfAnyLookup<TNegator, TOptimizations>(Vector128<short> source0, Vector128<short> source1, Vector128<byte> bitmapLookup)
+            where TNegator : struct, INegator
+            where TOptimizations : struct, IOptimizations
+        {
+            // Pack two vectors of characters into bytes.
+            // X86: Downcast every character using saturation.
+            // - Values <= 32767 result in min(value, 255).
+            // - Values  > 32767 result in 0. Because of this we must do more work to handle needles that contain 0.
+            // ARM64: Take the low byte of each character.
+            // - All values result in (value & 0xFF).
+            Vector128<byte> source = Sse2.IsSupported
+                ? Sse2.PackUnsignedSaturate(source0, source1)
```

Comment:
I'm not sure that gives you anything though, just shifts the problem to now having to explicitly handle `127`.

Reply:
Might be, but you have to check 0 regardless because of the >= 32768 range where `PackSignedSaturate` will produce values > 127 and Shuffle will spit out 0s again.
Unless I'm missing some other trick?

Reply:
> \> 127 and Shuffle will spit out 0s again

Ahhh, forgot about this. I'm sure there's some trick for this...I have to work on Sunday on something similar, so if I find something I'll let you know.